src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -797,6 +800,7 @@ private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<Symbol
         /// <param name="typeArguments">Containing symbol type arguments.</param>
         /// <param name="currentCompilation">Improves error message detail.</param>
         /// <param name="diagnosticsBuilder">Diagnostics.</param>
+        /// <param name="warningsBuilderOpt">Nullability warnings.</param>
```

Comment:
>warningsBuilderOpt [](start = 25, length = 18)

Why do we need to keep warnings separate from the rest of diagnostics?

Reply:
To allow `NullableWalker` to drop all but the nullability diagnostics.

---
In reply to: [207376397](https://github.com/dotnet/roslyn/pull/28959#discussion_r207376397) [](ancestors = 207376397)